tasks/kmt.py

```diff
@@ -284,24 +285,34 @@ def filter_target_domains(vms: str, infra: Dict[ArchOrLocal, HostInstance], loca
     return domains
 
 
+def get_archs_in_domains(domains: Iterable[LibvirtDomain]) -> Set[Arch]:
+    archs: Set[Arch] = set()
+    for d in domains:
+        archs.add(full_arch(d.instance.arch))
+    return archs
+
+
 TOOLS_PATH = '/datadog-agent/internal/tools'
 GOTESTSUM = "gotest.tools/gotestsum"
 
 
-def download_gotestsum(ctx: Context):
+def download_gotestsum(ctx: Context, arch: Arch):
     fgotestsum = "./test/kitchen/site-cookbooks/dd-system-probe-check/files/default/gotestsum"
+
     if os.path.isfile(fgotestsum):
-        return
+        file_arch = get_binary_target_arch(ctx, fgotestsum)
+        if file_arch == arch:
+            return
 
-    if not os.path.exists("kmt-deps/tools"):
-        ctx.run("mkdir -p kmt-deps/tools")
+    paths = KMTPaths(None, arch)
+    paths.tools.mkdir(parents=True, exist_ok=True)
 
-    docker_exec(
-        ctx,
-        f"cd {TOOLS_PATH} && go install {GOTESTSUM} && cp /go/bin/gotestsum /datadog-agent/kmt-deps/tools/",
+    cc = get_compiler(ctx, arch)
+    cc.exec(
+        f"cd {TOOLS_PATH} && go install {GOTESTSUM} && cp /go/bin/gotestsum /datadog-agent/{paths.tools}",
```

Comment:
This is failing on linux
```
docker exec -u compiler -i kmt-compiler-x86_64 bash -c "cd /datadog-agent/internal/tools && go install gotest.tools/gotestsum && cp /go/bin/gotestsum /datadog-agent//home/usama.saqib/go/github.com/DataDog/datadog-agent/kmt-deps/x86_64/tools"
cp: cannot create regular file '/datadog-agent//home/usama.saqib/go/github.com/DataDog/datadog-agent/kmt-deps/x86_64/tools': No such file or directory
```

Reply:
Fixed, I made the kmt-deps path always relative to the repo root instead of the current working directory and didn't notice that in this case it was used with another root. Thanks for the catch!